File: iOS/DateMate/UI/Auth/SignUp/SignUpUiState.swift
//
//  SignUpUiState.swift
//  DateMate
//
//  Created by 홍희표 on 2025/11/25.
//

struct SignUpUiState {
    var email: String = ""
    var password: String = ""
    var confirmPassword: String = ""
    var emailError: String? = nil
    var passwordError: String? = nil
    var confirmPasswordError: String? = nil
    var isLoading: Bool = false
    var isSignUpSuccess: Bool = false
    var errorMessage: String? = nil
    var isSignUpEnabled: Bool {
        return !email.isEmpty && emailError == nil &&
            !password.isEmpty && passwordError == nil &&
            !confirmPassword.isEmpty && confirmPassword == password
    }
}

File: Android/app/src/main/java/com/hhp227/datemate/ui/auth/signup/SignUpUiState.kt
package com.hhp227.datemate.ui.auth.signup

data class SignUpUiState(
    val email: String = "",
    val password: String = "",
    val confirmPassword: String = "",
    val emailError: String? = null,
    val passwordError: String? = null,
    val confirmPasswordError: String? = null,
    val isLoading: Boolean = false,
    val isSignUpSuccess: Boolean = false,
    val errorMessage: String? = null
) {
    val isSignUpEnabled: Boolean
        get() = email.isNotBlank() && emailError == null &&
                password.isNotBlank() && passwordError == null &&
                confirmPassword.isNotBlank() && confirmPassword == password
}